logRequests.go

```diff
@@ -1,30 +1,42 @@
 package transport
 
 import (
-	"log"
 	"net/http"
 	"time"
+	"log"
 
 	"moul.io/http2curl/v2"
 )
 
-func LogRequests(next http.RoundTripper) http.RoundTripper {
-	return RoundTripFunc(func(req *http.Request) (resp *http.Response, err error) {
-		r := cloneRequest(req)
-
-		curlCommand, _ := http2curl.GetCurlCommand(r)
-		log.Printf("%v", curlCommand)
-		log.Printf("request: %s %s", r.Method, r.URL)
-
-		startTime := time.Now()
-		defer func() {
-			if resp != nil {
-				log.Printf("response (HTTP %v): %v %s", time.Since(startTime), resp.Status, r.URL)
-			} else {
-				log.Printf("response (<nil>): %v %s", time.Since(startTime), r.URL)
-			}
-		}()
-
-		return next.RoundTrip(r)
-	})
+type DefaultLogger struct{}
+
+func (*DefaultLogger) Info(format string, v ...interface{}) {
+	log.Printf(format, v...)
+}
+
+func LogRequests(logger Logger) func(http.RoundTripper) http.RoundTripper {
+	if logger == nil {
+		logger = &DefaultLogger{}
+	}
+
+	return func(next http.RoundTripper) http.RoundTripper {
+		return RoundTripFunc(func(req *http.Request) (resp *http.Response, err error) {
+			r := cloneRequest(req)
+
+			curlCommand, _ := http2curl.GetCurlCommand(r)
+			logger.Info("%v", curlCommand)
+			logger.Info("request: %s %s", r.Method, r.URL)
+
+			startTime := time.Now()
+			defer func() {
+				if resp != nil {
+					logger.Info("response (HTTP %v): %v %s", time.Since(startTime), resp.Status, r.URL)
+				} else {
+					logger.Info("response (<nil>): %v %s", time.Since(startTime), r.URL)
+				}
+			}()
+
+			return next.RoundTrip(r)
+		})
+	}
 }
```

Comment:
I like this.. however instead of Logger can we just use io.Writer?

I found myself needing this, so I could save each request to a buffer and only log/print it on error :)

Reply:
I totally forgot that I created this PR and today I created similar PR just week after :D It's the heat I guess.. 


Anyway what do you think about this approach: https://github.com/go-chi/transport/pull/7 ???

This way the consumer of the package can do whatever he likes